homeassistant/components/panasonic_viera/media_player.py

```diff
@@ -168,79 +168,136 @@ def supported_features(self):
             self._wol.send_magic_packet(self._mac, ip_address=self._broadcast)
             self._state = STATE_ON
         elif self._app_power:
-            self._remote.turn_on()
-            self._state = STATE_ON
+            if self._remote:
+                self._remote.turn_on()
+                self._state = STATE_ON
 
     def turn_off(self):
         """Turn off media player."""
-        if self._state != STATE_OFF:
+        if self._remote and self._state != STATE_OFF:
             self._remote.turn_off()
             self._state = STATE_OFF
 
     def volume_up(self):
         """Volume up the media player."""
-        self._remote.volume_up()
+        if self._remote:
+            self._remote.volume_up()
 
     def volume_down(self):
         """Volume down media player."""
-        self._remote.volume_down()
+        if self._remote:
+            self._remote.volume_down()
 
     def mute_volume(self, mute):
         """Send mute command."""
-        self._remote.set_mute(mute)
+        if self._remote:
+            self._remote.set_mute(mute)
 
     def set_volume_level(self, volume):
         """Set volume level, range 0..1."""
-        volume = int(volume * 100)
-        try:
-            self._remote.set_volume(volume)
-            self._state = STATE_ON
-        except OSError:
-            self._state = STATE_OFF
+        if self._remote:
+            volume = int(volume * 100)
+            try:
+                self._remote.set_volume(volume)
+                self._state = STATE_ON
+            except OSError:
+                self._state = STATE_OFF
 
     def media_play_pause(self):
         """Simulate play pause media player."""
-        if self._playing:
-            self.media_pause()
-        else:
-            self.media_play()
+        if self._remote:
+            if self._playing:
+                self.media_pause()
+            else:
+                self.media_play()
 
     def media_play(self):
         """Send play command."""
-        self._playing = True
-        self._remote.media_play()
+        if self._remote:
+            self._playing = True
+            self._remote.media_play()
 
     def media_pause(self):
         """Send media pause command to media player."""
-        self._playing = False
-        self._remote.media_pause()
+        if self._remote:
+            self._playing = False
+            self._remote.media_pause()
 
     def media_next_track(self):
         """Send next track command."""
-        self._remote.media_next_track()
+        if self._remote:
+            self._remote.media_next_track()
 
     def media_previous_track(self):
         """Send the previous track command."""
-        self._remote.media_previous_track()
+        if self._remote:
+            self._remote.media_previous_track()
 
     def play_media(self, media_type, media_id, **kwargs):
         """Play media."""
-        _LOGGER.debug("Play media: %s (%s)", media_id, media_type)
+        if self._remote:
```

Comment:
Invert this check and return if true.

Reply:
Invert it how? Placing `_LOGGER.debug("Play media: %s (%s)", media_id, media_type)` before the if statement?

Reply:
```py
        if not self._remote:
            return
        _LOGGER.debug("Play media: %s (%s)", media_id, media_type)
        ...
```

Reply:
This pattern is called a guard clause. It saves on indentation.

Reply:
Oh, great!